### ISSUE TITLE
[Android] Maintain current (Android < 35) `SystemUiMode` behavior on Android >= 35

### ISSUE THREAD

**Issue by camsim99** [flutter/flutter#150216]
According to [docs](https://developer.android.com/about/versions/15/behavior-changes-15#edge-to-edge), edge-to-edge will become the default layout mode for apps targeting Android 15/35 or later. In order to avoid completely breaking Flutter apps once they start targeting SDK 35+ or run on Android 15 devices, we need to make sure that the current non-edge-to-edge [system UI modes](https://api.flutter.dev/flutter/services/SystemUiMode.html) work the same as before. This means that for those system UI modes, we must ensure that we disable edge-to-edge.

Additionally, we should change [our logic for opting into edge-to-edge](https://github.com/flutter/engine/blob/3023a964533846a8fcf3f044a7c2496fa91b251f/shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java#L312) to use the recommended [EdgeToEdge](https://developer.android.com/reference/androidx/activity/EdgeToEdge#(androidx.activity.ComponentActivity).enable(androidx.activity.SystemBarStyle,androidx.activity.SystemBarStyle)) API versus setting deprecated window flags ourselves; see the [official recommendation](https://developer.android.com/develop/ui/views/layout/edge-to-edge).

**Comment by camsim99** [comment]
Note to self: Start with changing Flutter apps to disable edge-to-edge mode by default then try to use the updated way to use edge-to-edge mode.

**Comment by camsim99** [comment]
https://developer.android.com/reference/android/R.attr#windowOptOutEdgeToEdgeEnforcement is what we'll need to opt Flutter apps out of edge-to-edge but this will require a check that API 35 is the SDK in use.

**Comment by camsim99** [comment]
Testing for this step should involve writing Golden tests to ensure that the behavior is what we expect before/after this change.

**Comment by camsim99** [comment]
Going to focus on writing tests for the System UI modes first because using Android styles in the Flutter engine has turned out to be a complicated task that should not block making sure we are prepared for the Android change in terms of tests and documentation (https://github.com/flutter/website/issues/10755).

**Comment by camsim99** [comment]
We determined that neither maintaining the current behavior nor writing goldens test to test the `SystemUiMode`s is possible.

We cannot maintain the current behavior, because we must create a style that the engine can reference to opt out of edge-to-edge by default, and this is not possible with the way the engine is currently compiled and used as a dependency in the framework. This work could be done, but we determined it is out of scope for this particular edge-to-edge work.

We cannot write goldens tests because golden tests + integration tests currently does not work. It is a WIP though by @/matanlurey I think (https://github.com/flutter/flutter/issues/143299), so this piece may be addressable in the future. Filed https://github.com/flutter/flutter/issues/153970 to track.
